app/models/school.rb

```diff
@@ -49,8 +59,8 @@ class School < ApplicationRecord
     where(id: Partnership.joins(:cohort).where(cohorts: { start_year: year }).select(:school_id))
   }
 
-  scope :partnered_with_lead_provider, lambda { |lead_provider_id|
-    where(id: Partnership.where(lead_provider_id: lead_provider_id).select(:school_id))
+  scope :partnered_with_lead_provider, lambda { |lead_provider_id, year|
+    where(id: Partnership.where(lead_provider_id: lead_provider_id).joins(:cohort).where(cohorts: { start_year: year }).select(:school_id))
```

Comment:
Could you use your new scope `for_year` here?

Reply:
Thanks, I knew I'd written that scope for a reason